api-rest/src/test/java/com/homihq/db2rest/jdbc/rest/meta/schema/SchemaControllerTest.java

```diff
@@ -0,0 +1,100 @@
+        List<TableObject> actualTableObject = schemaController.getObjects("key",null);
+        List<TableObject> expectedTableObject = metadataMap.get("key").dbTables().stream().map(t-> new TableObject(t.schema(),t.name(),t.type())).toList();
+        schemaController.getObjects("key",null);
+        assertEquals(expectedTableObject,actualTableObject);
+    }
+    @Test
+    void testGetObjectsFilterInvalid() {
+        when(jdbcManager.getDbMetaByDbId("key")).thenReturn(metadataMap.get("key"));
+        Throwable exception = assertThrows(GenericDataAccessException.class,() -> schemaController.getObjects("key","invalidFilter"));
+        assertEquals("Invalid filter condition. Only == supported for schema filter using a single value only.",exception.getMessage());
+    }
+    @Test
+    void testGetObjectsValidFilter_schema(){
+        when(jdbcManager.getDbMetaByDbId("key")).thenReturn(metadataMap.get("key"));
+        List<TableObject> expectedTableObject = metadataMap.get("key").dbTables().stream().map(t-> new TableObject(t.schema(),t.name(),t.type())).toList();
+        List<TableObject> actualTableObject = schemaController.getObjects("key","schema==schema");
+        assertEquals(expectedTableObject,actualTableObject);
+
+        actualTableObject = schemaController.getObjects("key","schema==notPresentInList");
+        assertEquals(0,actualTableObject.size());
+    }
+
+    @Test
+    void testGetObjectsValidFilter_name() {
+        when(jdbcManager.getDbMetaByDbId("key")).thenReturn(metadataMap.get("key"));
+        List<TableObject> expectedTableObject = metadataMap.get("key").dbTables().stream().map(t-> new TableObject(t.schema(),t.name(),t.type())).toList();
+        List<TableObject> actualTableObject = schemaController.getObjects("key","name==name");
+        assertEquals(expectedTableObject,actualTableObject);
+
+        actualTableObject = schemaController.getObjects("key","name==notPresentInList");
+        assertEquals(0,actualTableObject.size());
+    }
+    @Test
+    void testGetObjectsValidFilter_type() {
+        when(jdbcManager.getDbMetaByDbId("key")).thenReturn(metadataMap.get("key"));
+        List<TableObject> expectedTableObject = metadataMap.get("key").dbTables().stream().map(t-> new TableObject(t.schema(),t.name(),t.type())).toList();
+        List<TableObject> actualTableObject = schemaController.getObjects("key","type==type");
+        assertEquals(expectedTableObject,actualTableObject);
+
+        actualTableObject = schemaController.getObjects("key","type==notPresentInList");
+        assertEquals(0,actualTableObject.size());
+    }
+    @Test
+    void testGetObjectsValidFilter_invalidFieldFilter() {
+        when(jdbcManager.getDbMetaByDbId("key")).thenReturn(metadataMap.get("key"));
+        List<TableObject> actualTableObject;
+        actualTableObject = schemaController.getObjects("key","invalidFilterField==notPresentInList");
+        assertEquals(0,actualTableObject.size());
```

Comment:
Hmm, that looks awkward.  We filter against a field, it's not found, and so we return an empty list.
@kdhrubo Should we improve the code to return a simple message such as "No matches found."?  Or is empty List enough?  If empty List is assumed to be the same as "No matches found.", then we should change this test result to not check a size, but ensure what actually is being returned with `assertEquals(actualTableObject.isEmpty())`

Some say "always return some object or message, never null".  Others say its fine for APIs to return nothing.  But even nothing is always represented by some object or message.  In this case, an empty List object.  So we should improve this.

Reply:
For now I have added both IT and UT for the current code itself. 
Do we have a plan for when this change will be done?